### PR TITLE
feat: add jwt auth and registration

### PR DESCRIPTION
**Pull request by matt-drayton**
feat: add jwt auth and registration

This PR adds JWT auth, as well as user views (registration, login, logout) to the project. It also configures the Axios proxy to allow requests to be made to the Django server from the frontend. 

Adds an unstyled and ugly login component :)

### CODE REVIEW

File: backend/backend/settings.py [hunk 1]
```diff
@@ -45,7 +42,8 @@
 
 MIDDLEWARE = [
     "django.middleware.security.SecurityMiddleware",
-    "django.contrib.sessions.middleware.SessionMiddleware",
+    "django_session_jwt.middleware.SessionMiddleware",
+    # "django.contrib.sessions.middleware.SessionMiddleware",
```
Comment (lloydtao) [inline backend/backend/settings.py]: remove redundant code comment

File: backend/backend/urls.py [hunk 2]
```diff
@@ -22,13 +22,16 @@
...
+from accounts.views import LoginView, LogoutView
 
 router = routers.DefaultRouter()
 router.registry.extend(accounts_router.registry)
 router.registry.extend(program_router.registry)
 router.registry.extend(schedule_router.registry)
 
 urlpatterns = [
+    path("api/login/", LoginView.as_view(), name="login"),
+    path("api/logout/", LogoutView.as_view(), name="logout"),
```
Comment (lloydtao) [inline backend/backend/urls.py:35]: put urls in a logical order (i.e. alphabetical)

**Review by lloydtao — CHANGES_REQUESTED** [review]
fix

File: backend/backend/settings.py (hunk 1 above)
Comment (matt-drayton) [reply backend/backend/settings.py]: https://github.com/LloydTao/Hackathon-NASA-2020-II/commit/42c1f1f4968028d352a9a57c6637892eee8ffca0 fixd

File: backend/backend/urls.py (hunk 2 above)
Comment (matt-drayton) [reply backend/backend/urls.py:35]: https://github.com/LloydTao/Hackathon-NASA-2020-II/commit/42c1f1f4968028d352a9a57c6637892eee8ffca0 fixd

File: backend/backend/settings.py (hunk 1 above)
Comment (lloydtao) [reply backend/backend/settings.py]: lgtm

File: backend/backend/urls.py (hunk 2 above)
Comment (lloydtao) [reply backend/backend/urls.py:35]: lgtm

**Review by lloydtao — APPROVED** [review]
lgtm

**Pull request merged**